src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java

```diff
@@ -67,6 +71,49 @@
 		return map;
 	}
 
+	protected PodSpecBuilder createPodSpecBuilder(KubernetesDeployerProperties properties) {
+		PodSpecBuilder podSpecBuilder = new PodSpecBuilder();
+
+		// Add image secrets if set
+		if (properties.getImagePullSecret() != null) {
+			podSpecBuilder.addNewImagePullSecret(properties.getImagePullSecret());
+		}
+
+		return podSpecBuilder;
+	}
+
+	protected Container addMemoryCpu(AppDeploymentRequest request, Container container,
+	                                 KubernetesDeployerProperties properties) {
+		// add memory and cpu resource limits
+		ResourceRequirements resourceRequirements = new ResourceRequirements();
+		resourceRequirements.setLimits(deduceResourceLimits(properties, request));
+		resourceRequirements.setRequests(deduceResourceRequests(properties, request));
+		container.setResources(resourceRequirements);
+		ImagePullPolicy pullPolicy = deduceImagePullPolicy(properties, request);
+		container.setImagePullPolicy(pullPolicy.name());
+		return container;
+	}
+
+	protected void addVolumes(PodSpecBuilder podSpecBuilder, KubernetesDeployerProperties properties) {
+		// Add volumes
+		podSpecBuilder.withVolumes(properties.getHostVolumeMounts().stream()
```

Comment:
also not sure if there is any relevance between `properties.getHostVolumeMounts()` and `container` volumeMounts. If there is any, in order for `properties.getHostVolumeMounts()` to get the correct volume mounts, should the above be called before creating the container?